accord-core/src/main/java/accord/topology/TopologyManager.java

```diff
@@ -0,0 +1,325 @@
+package accord.topology;
+
+import accord.api.ConfigurationService;
+import accord.coordinate.tracking.QuorumTracker;
+import accord.local.Node;
+import accord.messages.Request;
+import accord.messages.TxnRequest;
+import accord.messages.TxnRequestScope;
+import accord.txn.Keys;
+import accord.txn.Txn;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+import java.util.*;
+import java.util.function.LongConsumer;
+
+/**
+ * Manages topology state changes and update bookkeeping
+ *
+ * Each time the topology changes we need to:
+ * * confirm previous owners of ranges we replicate are aware of the new config
+ * * learn of any outstanding operations for ranges we replicate
+ * * clean up obsolete data
+ *
+ * Assumes a topology service that won't report epoch n without having n-1 etc also available
+ */
+public class TopologyManager implements ConfigurationService.Listener
+{
+    class EpochState
+    {
+        private final Topology global;
+        private final Topology local;
+        private final QuorumTracker syncTracker;
+        private boolean syncComplete = false;
+        private boolean prevSynced;
+
+        private boolean updateState(boolean prevSynced)
+        {
+            this.prevSynced = prevSynced;
+            if (!syncComplete)
+            {
+                syncComplete = syncTracker.hasReachedQuorum();
+                return syncComplete;
```

Comment:
Should this be `prevSynced && syncComplete`? Or perhaps we should call the method `onPrevSynced()` so that we cannot provide a value of `false`?